WordPress/gutenberg
# Decode some characters if used in taxonomy name so it's displayed correctly in Query Loop filters

## What?
A couple of characters are not decoded properly and displayed encoded:
- `&` -> `&amp;`
- `>` -> `&gt;`
- `<` -> `&lt;`

This PR solves this problem.

## Why?
Users should see decoded values in the frontend.

## How?
Values provided to FormTokenField are now decoded using `decodeEntities`. Decoding happens only in presentation layer.

## Testing Instructions
1. Create a new post category (Posts -> Categories) with a name that includes `&`, `<`, `>` e.g. `Sports & Games > winter <`
2. Assign a post to this new category
3. Create a new post
4. Include `Query Loop`
5. Add Categories filter (Filters -> + -> Taxonomies)
6. Type "Sports"
7. Expected: `Sports & Games > winter <` is displayed as a suggestion (Not `Sports &amp; Games &gt; WINTER &lt;`)
8. Click on this category
9. Expected: `Sports & Games > winter <` is displayed as a chosen option (Not `Sports &amp; Games`) and posts are filtered correctly

## Screenshots or screencast 

| Before | After |
| --- | --- |
| <img width="276" alt="image" src="https://github.com/WordPress/gutenberg/assets/20098064/860bc920-b0ac-4c32-b6e0-4138e55b8fa9"> | <img width="291" alt="image" src="https://github.com/WordPress/gutenberg/assets/20098064/7df5e092-5abe-48a8-9c33-bf3ba1508b49"> |

Note: Using the `<` and `>` characters in the title in this order makes the content disappear. For example:
- `Sports & Games <winter>` becomes `Sports & Games` but that's not in the scope of this PR.
